Issue on python 2.7.5 running unittests

When running unittests, on python  2.7.5, I get some failures.

I'm not running virtualenv

<pre>
[myork@z820 nailgun]$ cat /etc/redhat-release 
Red Hat Enterprise Linux Workstation release 7.1 (Maipo)
</pre>

<pre>
>>> print sys.version
2.7.5 (default, Feb 11 2014, 07:46:25) 
[GCC 4.8.2 20140120 (Red Hat 4.8.2-13)]
</pre>

Unittests are all run from within pycharm.

<pre>
/usr/bin/python2.7 /home/myork/applications/pycharm-4.5.1/helpers/pycharm/utrunner.py /home/myork/PycharmProjects/nailgun/ true
Testing started at 15:02 ...

Failure
Expected :"test_entity_mixins.SampleEntityTwo(nailgun.config.ServerConfig(url='http://example.com'))"
Actual   :"tests.test_entity_mixins.SampleEntityTwo(nailgun.config.ServerConfig(url='http://example.com'))"
 <Click to see difference>

Traceback (most recent call last):
  File "/home/myork/PycharmProjects/nailgun/tests/test_entity_mixins.py", line 311, in test_repr_v1
    self.assertEqual(repr(entity), target)
AssertionError: "test_entity_mixins.SampleEntityTwo(nailgun.config.ServerConfig(url='http://example.com'))" != "tests.test_entity_mixins.SampleEntityTwo(nailgun.config.ServerConfig(url='http://example.com'))"


Failure
Expected :"test_entity_mixins.SampleEntityTwo(nailgun.config.ServerConfig(url='http://example.com'), id=2495229732896996198)"
Actual   :"tests.test_entity_mixins.SampleEntityTwo(nailgun.config.ServerConfig(url='http://example.com'), id=2495229732896996198)"
 <Click to see difference>

Traceback (most recent call last):
  File "/home/myork/PycharmProjects/nailgun/tests/test_entity_mixins.py", line 329, in test_repr_v2
    self.assertEqual(repr(entity), target)
AssertionError: "test_entity_mixins.SampleEntityTwo(nailgun.config.ServerConfig(url='http://example.com'), id=2495229732896996198)" != "tests.test_entity_mixins.SampleEntityTwo(nailgun.config.ServerConfig(url='http://example.com'), id=2495229732896996198)"


Failure
Expected :"test_entity_mixins.SampleEntityTwo(nailgun.config.ServerConfig(url='http://example.com'), one_to_many=[test_entity_mixins.SampleEntity(nailgun.config.ServerConfig(url='http://example.com'), id=3965265452630526438)])"
Actual   :"tests.test_entity_mixins.SampleEntityTwo(nailgun.config.ServerConfig(url='http://example.com'), one_to_many=[tests.test_entity_mixins.SampleEntity(nailgun.config.ServerConfig(url='http://example.com'), id=3965265452630526438)])"
 <Click to see difference>

Traceback (most recent call last):
  File "/home/myork/PycharmProjects/nailgun/tests/test_entity_mixins.py", line 354, in test_repr_v3
    self.assertEqual(repr(entity), target)
AssertionError: "test_entity_mixins.SampleEntityTwo(nailgun.config.ServerConfig(url='http://example.com'), one_to_many=[test_entity_mixins.SampleEntity(nailgun.config.ServerConfig(url='http://example.com'), id=3965265452630526438)])" != "tests.test_entity_mixins.SampleEntityTwo(nailgun.config.ServerConfig(url='http://example.com'), one_to_many=[tests.test_entity_mixins.SampleEntity(nailgun.config.ServerConfig(url='http://example.com'), id=3965265452630526438)])"


Failure
Traceback (most recent call last):
  File "/home/myork/PycharmProjects/nailgun/tests/test_config.py", line 74, in test_get
    _compare_configs(self, config, server_config)
  File "/home/myork/PycharmProjects/nailgun/tests/test_config.py", line 44, in _compare_configs
    self.assertEqual(dict_config, cfg)
AssertionError: {'url': 'booger', 'version': '1.2.3.4.dev5+67'} != {'url': u'booger', 'version': "(u'00000001', u'00000002', u'00000003', u'0000000 [truncated]...
- {'url': 'booger', 'version': '1.2.3.4.dev5+67'}
+ {'url': u'booger',
+  'version': "(u'00000001', u'00000002', u'00000003', u'00000004', '*@', u'00000005', u'*+', u'00000067', '*final')"}


Failure
Traceback (most recent call last):
  File "/home/myork/PycharmProjects/nailgun/tests/test_config.py", line 58, in test_init
    _compare_configs(self, config, server_config)
  File "/home/myork/PycharmProjects/nailgun/tests/test_config.py", line 44, in _compare_configs
    self.assertEqual(dict_config, cfg)
AssertionError: {'url': 'booger', 'version': '1.2.3.4.dev5+67'} != {'url': 'booger', 'version': "('00000001', '00000002', '00000003', '00000004', ' [truncated]...
- {'url': 'booger', 'version': '1.2.3.4.dev5+67'}
+ {'url': 'booger',
+  'version': "('00000001', '00000002', '00000003', '00000004', '*@', '00000005', '*+', '00000067', '*final')"}


Failure
Traceback (most recent call last):
  File "/home/myork/PycharmProjects/nailgun/tests/test_config.py", line 150, in test_init
    _compare_configs(self, config, ServerConfig(**config))
  File "/home/myork/PycharmProjects/nailgun/tests/test_config.py", line 44, in _compare_configs
    self.assertEqual(dict_config, cfg)
AssertionError: {'url': 'booger', 'version': '1.2.3.4.dev5+67'} != {'url': 'booger', 'version': "('00000001', '00000002', '00000003', '00000004', ' [truncated]...
- {'url': 'booger', 'version': '1.2.3.4.dev5+67'}
+ {'url': 'booger',
+  'version': "('00000001', '00000002', '00000003', '00000004', '*@', '00000005', '*+', '00000067', '*final')"}


Process finished with exit code 0
</pre>

I've just checked, and pycharm does have it's own test runner, I had assumed (incorrectly) the test results would be the same.

In fact there are only 3 errors when I use <pre>make test</pre>

Here is the output from make test

<pre>
[myork@z820 nailgun]$ make test
python -m unittest discover --start-directory tests --top-level-directory .
....................................................................................................F.F..........F
======================================================================
FAIL: test_get (tests.test_config.BaseServerConfigTestCase)
Test :meth:`nailgun.config.BaseServerConfig.get`.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_config.py", line 74, in test_get
    _compare_configs(self, config, server_config)
  File "tests/test_config.py", line 44, in _compare_configs
    self.assertEqual(dict_config, cfg)
AssertionError: {'url': 'booger', 'version': '1.2.3.4.dev5+67'} != {'url': u'booger', 'version': "(u'00000001', u'00000002', u'00000003', u'0000000 [truncated]...
- {'url': 'booger', 'version': '1.2.3.4.dev5+67'}
+ {'url': u'booger',
+  'version': "(u'00000001', u'00000002', u'00000003', u'00000004', '*@', u'00000005', u'*+', u'00000067', '*final')"}

======================================================================
FAIL: test_init (tests.test_config.BaseServerConfigTestCase)
Test instantiating :class:`nailgun.config.BaseServerConfig`.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_config.py", line 58, in test_init
    _compare_configs(self, config, server_config)
  File "tests/test_config.py", line 44, in _compare_configs
    self.assertEqual(dict_config, cfg)
AssertionError: {'url': 'booger', 'version': '1.2.3.4.dev5+67'} != {'url': 'booger', 'version': "('00000001', '00000002', '00000003', '00000004', ' [truncated]...
- {'url': 'booger', 'version': '1.2.3.4.dev5+67'}
+ {'url': 'booger',
+  'version': "('00000001', '00000002', '00000003', '00000004', '*@', '00000005', '*+', '00000067', '*final')"}

======================================================================
FAIL: test_init (tests.test_config.ServerConfigTestCase)
Test instantiating :class:`nailgun.config.ServerConfig`.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_config.py", line 150, in test_init
    _compare_configs(self, config, ServerConfig(**config))
  File "tests/test_config.py", line 44, in _compare_configs
    self.assertEqual(dict_config, cfg)
AssertionError: {'url': 'booger', 'version': '1.2.3.4.dev5+67'} != {'url': 'booger', 'version': "('00000001', '00000002', '00000003', '00000004', ' [truncated]...
- {'url': 'booger', 'version': '1.2.3.4.dev5+67'}
+ {'url': 'booger',
+  'version': "('00000001', '00000002', '00000003', '00000004', '*@', '00000005', '*+', '00000067', '*final')"}

----------------------------------------------------------------------
Ran 114 tests in 0.139s

FAILED (failures=3)
make: *** [test] Error 1
</pre>

I will checkout the latest branch and test it again.

--- a/tests/test_config.py
+++ b/tests/test_config.py
@@ -38,10 +38,10 @@ def _compare_configs(self, dict_config, server_config):
         subclass thereof.
 
     """
-    cfg = vars(server_config).copy()
-    if 'version' in cfg:
-        cfg['version'] = str(cfg['version'])
-    self.assertEqual(dict_config, cfg)
+    if 'version' in dict_config:
+        dict_config = dict_config.copy()  # shadow the passed in dict
+        dict_config['version'] = parse_version(dict_config.pop('version'))
+    self.assertEqual(dict_config, vars(server_config))
 
 
 class BaseServerConfigTestCase(TestCase):

